src/itemstackmetadata.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "itemstackmetadata.h"
+#include "exceptions.h"
+#include "gamedef.h"
+#include "inventory.h"
+#include "log.h"
+#include "util/serialize.h"
+#include "util/strfnd.h"
+#include "constants.h" // MAP_BLOCKSIZE
+#include <sstream>
+
+/*
+	ItemStackMetadata
+*/
+
+std::string ItemStackMetadata::serialize() const
+{
+	std::ostringstream os;
+	os << "$";
+	for (StringMap::const_iterator
+			it = m_stringvars.begin();
+			it != m_stringvars.end(); ++it) {
+		os << it->first << "=";
+		os << serializeJsonString(it->second) << "\n";
+	}
+	return os.str();
+}
+
+void ItemStackMetadata::deSerialize(std::string &in)
+{
+	m_stringvars.clear();
+	//to indicate new serialisation version, a dollar sign has to be the first char in the serialized string. if not, apply legacy deserialization
+	if (in[0] == '$') {
```

Comment:
How about using ASCII control codes, like `#define DESERIALIZE_START 0x01` and `#define DESERIALIZE_DELIM 0x02` defined in itemstackmetadata.h to allow the both characters in the key and/or value?